overlord/ifacestate/hotplug.go

```diff
@@ -86,14 +106,210 @@ func defaultDeviceKey(devinfo *hotplug.HotplugDeviceInfo, keyVersion int) (strin
 
 // hotplugDeviceAdded gets called when a device is added to the system.
 func (m *InterfaceManager) hotplugDeviceAdded(devinfo *hotplug.HotplugDeviceInfo) {
+	st := m.state
+	st.Lock()
+	defer st.Unlock()
+
+	if _, err := snapstate.CoreInfo(st); err != nil {
+		logger.Noticef("core snap not available, hotplug events ignored")
+		return
+	}
+
+	hotplugIfaces := m.repo.AllHotplugInterfaces()
```

Comment:
wouldn't it be more convenient if this returned a set ( map to bool ) directly?

Reply:
Indeed, good idea, thanks!